app/controllers/vacancies_controller.rb

```diff
@@ -25,6 +27,16 @@ def show
     @vacancy = Vacancy.friendly.find(params[:id])
   end
 
+  def edit
+    @vacancy
+  end
+
+  def update
+    @vacancy = Vacancy.friendly.find(params[:id])
```

Comment:
I don't like this. You are getting the vacancy and checking if it belongs to the user.

You can change this by:
```
@vacancy = current_company.vacancies.friendly.find(params[:id])
```

.remove `validate_ownership`, `render_bad_ownership`, `VacancyDontBelongsToUser` class,

and rely on HTTP. It will return a 404 when the user tries to update a vacancy from another user.

Reply:
nice